dive_core/command_hierarchy.cpp

```diff
@@ -1764,14 +1773,59 @@ std::string CommandHierarchyCreator::GetEventString(const IMemoryManager &mem_ma
     return string_stream.str();
 }
 
+//--------------------------------------------------------------------------------------------------
+void CommandHierarchyCreator::AppendRegNodes(const IMemoryManager &mem_manager,
+                                             uint32_t              submit_index,
+                                             uint64_t              va_addr,
+                                             uint32_t              dword_count,
+                                             uint64_t              packet_node_index)
+{
+    // This version of AppendRegNodes takes in a raw buffer consisting of register offset + value
+    // pairs
+    DIVE_ASSERT(dword_count % 2 == 0);
+    for (uint32_t i = 0; i < dword_count; i += 2)
+    {
+        struct
+        {
+            uint32_t m_reg_offset;
+            uint32_t m_reg_value;
+        } reg_pair;
+        uint64_t pair_addr = va_addr + i * sizeof(uint32_t);
+        DIVE_VERIFY(mem_manager.CopyMemory(&reg_pair, submit_index, pair_addr, sizeof(reg_pair)));
+
+        const RegInfo *reg_info_ptr = GetRegInfo(reg_pair.m_reg_offset);
```

Comment:
maybe better with a `GetRegInfo` function since this code appears in the other `AppendRegNodes()` as well

Reply:
Actually the 2 AppendRegNodes() looks very similar, is it possible to combine the 2 into 1 function?

Reply:
They look similar but are different in fundamental ways. One expects a list of register values and has the offset passed in as a parameter, the other one expects (offset,value) pairs. There can be an attempt at extracting the common parts, but the 2 functions are short enough that I don't think the effort and additional indirection is worth it